File: app/src/main/java/com/stevenfrew/beatprompter/midi/WildcardValue.java
package com.stevenfrew.beatprompter.midi;

/**
 * Represents a value that matches anything else.
 */
public class WildcardValue extends Value {

    @Override
    byte resolve(byte[] arguments, byte channel) {
        return 0;
    }

    @Override
    boolean matches(Value otherValue) {
        return true;
    }

    @Override
    public String toString()
    {
        return "*";
    }
}

File: app/src/main/java/com/stevenfrew/beatprompter/midi/WildcardValue.kt
package com.stevenfrew.beatprompter.midi

/**
 * Represents a value that matches anything else.
 */
class WildcardValue : Value() {

    internal override fun resolve(arguments: ByteArray, channel: Byte): Byte {
        return 0
    }

    internal override fun matches(otherValue: Value?): Boolean {
        return true
    }

    override fun toString(): String {
        return "*"
    }
}